File: 374.guess-number-higher-or-lower/solution.java
/* The guess API is defined in the parent class GuessGame.
   @param num, your guess
   @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
      int guess(int num); */

public class Solution extends GuessGame {
    public int guessNumber(int n) {
        int lo = 1;
        int hi = n;
        while (lo <= hi) {
            int k = lo + (hi - lo) / 2;
            switch (guess(k)) {
                case -1:
                    hi = k -1;
                    break;
                case 0:
                    return k;
                case 1:
                    lo = k + 1;
                    break;
            }
        }
        return -1;
    }
}

File: 374.guess-number-higher-or-lower/solution.kt
/** 
 * The API guess is defined in the parent class.
 * @param  num   your guess
 * @return 	     -1 if num is lower than the guess number
 *			      1 if num is higher than the guess number
 *               otherwise return 0
 * fun guess(num:Int):Int {}
 */

class Solution:GuessGame() {
    override fun guessNumber(n:Int):Int {
        var low = 1
        var high = n
        while (low < high) {
            val mid = low + (high - low) / 2
            when(guess(mid)) {
                -1 -> high = mid - 1
                0 -> return mid
                1 -> low = mid + 1
            }
        }
        return low
    }
}